components/brave_vpn/browser/connection/win/brave_vpn_helper/brave_vpn_helper_crash_reporter_client.cc

```diff
@@ -0,0 +1,156 @@
+#include "chrome/install_static/install_util.h"
+#include "chrome/install_static/product_install_details.h"
+#include "chrome/install_static/user_data_dir.h"
+#include "components/crash/core/app/crash_switches.h"
+#include "components/crash/core/app/crashpad.h"
+#include "components/version_info/channel.h"
+
+BraveVPNHelperCrashReporterClient::BraveVPNHelperCrashReporterClient() =
+    default;
+
+BraveVPNHelperCrashReporterClient::~BraveVPNHelperCrashReporterClient() =
+    default;
+
+// static
+void BraveVPNHelperCrashReporterClient::InitializeCrashReportingForProcess(
+    const std::string& process_type) {
+  static BraveVPNHelperCrashReporterClient* instance = nullptr;
+  if (instance)
+    return;
+
+  instance = new BraveVPNHelperCrashReporterClient();
+  ANNOTATE_LEAKING_OBJECT_PTR(instance);
+  // Don't set up Crashpad crash reporting in the Crashpad handler itself, nor
+  // in the fallback crash handler for the Crashpad handler process.
+  if (process_type == crash_reporter::switches::kCrashpadHandler) {
+    return;
+  }
+  install_static::InitializeProductDetailsForPrimaryModule();
+  crash_reporter::SetCrashReporterClient(instance);
+  std::wstring user_data_dir;
+  install_static::GetUserDataDirectory(&user_data_dir, nullptr);
+
+  crash_reporter::InitializeCrashpadWithEmbeddedHandler(
+      true, brave_vpn::kBraveVPNHelperProcessType,
+      install_static::WideToUTF8(user_data_dir), base::FilePath());
+}
+
+bool BraveVPNHelperCrashReporterClient::ShouldCreatePipeName(
+    const std::wstring& process_type) {
+  return false;
+}
+
+bool BraveVPNHelperCrashReporterClient::GetAlternativeCrashDumpLocation(
+    std::wstring* crash_dir) {
+  return false;
+}
+
+void BraveVPNHelperCrashReporterClient::GetProductNameAndVersion(
+    const std::wstring& exe_path,
+    std::wstring* product_name,
+    std::wstring* version,
+    std::wstring* special_build,
+    std::wstring* channel_name) {
+  *product_name = brave_vpn::kBraveVpnServiceName;
+  std::unique_ptr<FileVersionInfo> version_info(
+      FileVersionInfo::CreateFileVersionInfo(base::FilePath(exe_path)));
+  if (version_info) {
+    *version = base::AsWString(version_info->product_version());
+    *special_build = base::AsWString(version_info->special_build());
+  } else {
+    *version = L"0.0.0.0-devel";
+    *special_build = std::wstring();
+  }
+
+  *channel_name =
+      install_static::GetChromeChannelName(/*with_extended_stable=*/true);
+}
+
+bool BraveVPNHelperCrashReporterClient::ShouldShowRestartDialog(
+    std::wstring* title,
+    std::wstring* message,
+    bool* is_rtl_locale) {
+  // There is no UX associated with brave_vpn_helper, so no dialog should be
+  // shown.
+  return false;
+}
+
+bool BraveVPNHelperCrashReporterClient::AboutToRestart() {
+  // The brave_vpn_helper should never be restarted after a crash.
+  return false;
+}
+
+bool BraveVPNHelperCrashReporterClient::GetIsPerUserInstall() {
+  return !install_static::IsSystemInstall();
```

Comment:
Q) Is it possible to run with this clien on non system install?

Reply:
it can be started in non-service mode with `--console` for debugging